main/autogen/autogen.cc

```diff
@@ -674,4 +676,331 @@ void ParsedFile::classlist(core::Context ctx, vector<string> &out) {
     }
 }
 
+NamedDefinition ParsedFile::toNamed(core::Context ctx, DefinitionRef def) {
+    auto names = showFullName(ctx, def);
+    vector<core::NameRef> parentName;
+    if (def.data(*this).parent_ref.exists()) {
+        auto parentRef = def.data(*this).parent_ref.data(*this);
+        if (!parentRef.resolved.empty()) {
+            parentName = parentRef.resolved;
+        } else {
+            parentName = parentRef.name;
+        }
+    }
+    return {def.data(*this), names, parentName, requires, tree.file};
+}
+
+void ParsedFile::addDefinitions(core::Context ctx, const AutoloaderConfig &alConfig, DefTree &root) {
```

Comment:
Should this be part of ParsedFile? 
Does it make sense to have something like ParsedFileBuilder?

Reply:
Moving to function:
`autogen::addAutoloaderDefinitions(core::Context, const AutoloaderConfig &, ParsedFile &, DefTree &)`